### PR TITLE
Clean up connection getters/setters for timeouts

### PR DESCRIPTION
**Pull request by markdirish**
For whatever reason, `connection.loginTimeout` had a getter _and_ a setter, when it needs to be set prior to connecting. Removed the setter, and also ensured that the getter returns whatever value was either set in the connection options, or if the driver has decided to change the option, the value that the driver changed it to.

I also removed the `connection.connectionTimeout` setter. It wasn't actually _doing_ anything, and although setting the connection timeout after connecting makes sense and is allowed, I'm not sure how best to handle the async nature of setting a new value in relation to JavaScript properties. Might need to implement a setter that is a function:

```
connection.setConnectionTimeout(value, callback?);
```
or maybe a generic one?
```
connection.setConnectionAttribute(attribute, value, callback?);
```

### CODE REVIEW

File: src/odbc.cpp [hunk 1]
```diff
@@ -336,17 +336,39 @@ class ConnectAsyncWorker : public ODBCAsyncWorker {
...
+        // data source, the driver substitutes that value and returns SQLSTATE
+        // 01S02 (Option value changed)."
+        if (return_code == SQL_SUCCESS_WITH_INFO)
+        {
+          return_code =
+          SQLGetConnectAttr
+          (
+            hDBC,
+            SQL_ATTR_LOGIN_TIMEOUT,
+            (SQLPOINTER) &options->loginTimeout,
```
Comment (kadler) [inline src/odbc.cpp:362]:
```suggestion
            (SQLPOINTER) (intptr_t) options->loginTimeout,
```

File: src/odbc.cpp [hunk 2]
```diff
@@ -325,7 +325,7 @@ class ConnectAsyncWorker : public ODBCAsyncWorker {
         return_code = SQLSetConnectAttr(
           hDBC,                                   // ConnectionHandle
           SQL_ATTR_CONNECTION_TIMEOUT,            // Attribute
-          (SQLPOINTER) size_t(options->connectionTimeout), // ValuePtr
+          (SQLPOINTER) intptr_t(options->connectionTimeout), // ValuePtr
```
Comment (kadler) [inline src/odbc.cpp:328]: Seems like an odd mix of casts. I guess technically the intptr_t is not a cast, but the constructor. Might be worth a macro INT2PTR or something.

**Review by kadler — COMMENTED** [review]
The SQLSetConnectAttr won't work as written, but otherwise looks good.

File: src/odbc.cpp (hunk 1 above)
Comment (kadler) [reply src/odbc.cpp:362]: Nevermind, I didn't see it was *get* not *set*.

**Review by kadler — APPROVED** [review]

File: src/odbc.cpp (hunk 2 above)
Comment (markdirish) [reply src/odbc.cpp:328]: Oops, I actually meant to do two casts. If I just cast to an SQLPOINTER, it complains that the 

> warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]

So if I cast to the correctly-sized int first, the fixes the warning. But if I just cast to `intptr_t` and not `SQLPOINTER` as well, I get an actual compiler error:

> error: invalid conversion from ‘intptr_t {aka long int}’ to ‘SQLPOINTER {aka void*}

**Pull request merged**